src/backend/distributed/utils/citus_clauses.c

```diff
@@ -46,6 +48,21 @@ RequiresMasterEvaluation(Query *query)
 		}
 	}
 
+	foreach(rteCell, query->rtable)
+	{
+		RangeTblEntry *rte = (RangeTblEntry *) lfirst(rteCell);
+
+		if (rte->rtekind != RTE_SUBQUERY)
```

Comment:
Good catch. We only support very trivial CTEs right now, but should still be supported and might otherwise slip through later.